### DISCUSSION TITLE
Does renovate need access to older commits / would a clone with --depth x help with big repos?

### DISCUSSION THREAD

**Discussion by herndlm** [renovatebot/renovate#15754] [Suggest an Idea]
I know there's https://github.com/renovatebot/renovate/issues/7840 already, but I was just wondering why renovate clones with the full history, would a simple [--depth](https://www.git-scm.com/docs/git-clone#Documentation/git-clone.txt---depthltdepthgt) limitation help here?
I'm sure I'm overlooking something, like rebases or so, that might make problems then? I just stumbled over a repo again that is extremely huge and seems to take ages to clone and I was looking for quick workarounds / improvements to opt-in.

**Comment by rarkins** [comment]
Renovate needs access to all its branches and an unknown depth of those, so we have switched intentionally.

**Reply by herndlm** [reply]
ah ok, thx

**Comment by viceice** [comment]
which platform are you using renovate against?

it should support blobless git clones, so renovate will only fetch the metadata and only the blobs for checkout the specific branches.

**Reply by herndlm** [reply]
bitbucket and gitlab, I think it does use it for gitlab for sure. But I did a manuel blobless clone of that affected repo and it was still really big and slow. A --depth 1 did work, but as rarkins wrote this is not possible then. Anyways, thx for the quick answers both of you!

**Reply by rarkins** [reply]
Do you think the slowness is from the depth of history or from the size of the repo today? We are also planning for partial clones where you can skip a test folder entirely, for example

**Reply by herndlm** [reply]
that particular repo was on bitbucket and when I asked the team owning it, they told me that there are a huge amount of branches that were never deleted. this might be the reason?
I was also told that renovate is not needed anymore there, so I could disable renovate for it and am happy again :)

**Reply by rarkins** [reply]
Bitbucket is a platform without the clone filtering so it does do a full clone unfortunately.
